src/index_intf.ml

```diff
@@ -268,6 +278,13 @@ module type Index = sig
           of consecutive operations, which can be specified by
           [sampling_interval]. If [sampling_interval] is not set, no operation
           is timed. *)
+
+      val sync_with_timer : t -> unit
+      (** Time sync operations. *)
+
+      val sync' : ?hook:[ `Before_offset_read ] Hook.t -> t -> unit
```

Comment:
can you expose `sync` instead of `sync`` ? Also do why do we need a `with_timer` and a non-`with-timer` one? Maybe always using the timers is a good idea?